katdal/test/test_lazy_indexer.py

```diff
@@ -89,3 +145,11 @@ def test_stage1_multiple_boolean_indices(self):
         stage1 = tuple([True] * d for d in self.data.shape)
         indexer = DaskLazyIndexer(self.data_dask, stage1)
         np.testing.assert_array_equal(indexer[:], self.data)
+
+    def test_stage2_multiple_boolean_indices(self):
+        stage1 = tuple([True] * d for d in self.data.shape)
+        indexer = DaskLazyIndexer(self.data_dask, stage1)
+        stage2 = tuple([True] * 2 + [False] * (d - 2) for d in self.data.shape)
```

Comment:
This is going to be reduced by the simplifier. Should maybe have a separate test case that won't be simplified to a non-fancy case.